util/gnmi_test.go

```diff
@@ -766,3 +766,61 @@ func TestFindModelData(t *testing.T) {
 		}
 	}
 }
+
+func TestJoinPaths(t *testing.T) {
+	tests := []struct {
+		desc                 string
+		prefix, suffix, want *gpb.Path
+		wantErrSubstring     string
+	}{{
+		desc:   "all empty",
+		prefix: &gpb.Path{},
+		suffix: &gpb.Path{},
+		want:   &gpb.Path{},
+	}, {
+		desc:   "prefix only",
+		prefix: &gpb.Path{Origin: "o", Target: "t", Elem: []*gpb.PathElem{{Name: "p"}}},
+		suffix: &gpb.Path{},
+		want:   &gpb.Path{Origin: "o", Target: "t", Elem: []*gpb.PathElem{{Name: "p"}}},
+	}, {
+		desc:   "suffix only",
+		prefix: &gpb.Path{},
+		suffix: &gpb.Path{Origin: "o", Target: "t", Elem: []*gpb.PathElem{{Name: "s"}}},
+		want:   &gpb.Path{Origin: "o", Target: "t", Elem: []*gpb.PathElem{{Name: "s"}}},
+	}, {
+		desc:   "elements joined",
+		prefix: &gpb.Path{Elem: []*gpb.PathElem{{Name: "p"}}},
+		suffix: &gpb.Path{Elem: []*gpb.PathElem{{Name: "s"}}},
+		want:   &gpb.Path{Elem: []*gpb.PathElem{{Name: "p"}, {Name: "s"}}},
+	}, {
+		desc:   "same origin and target",
+		prefix: &gpb.Path{Origin: "o", Target: "t"},
+		suffix: &gpb.Path{Origin: "o", Target: "t"},
+		want:   &gpb.Path{Origin: "o", Target: "t"},
+	}, {
+		desc:             "mismatch origins",
+		prefix:           &gpb.Path{Origin: "o1"},
+		suffix:           &gpb.Path{Origin: "o2"},
+		wantErrSubstring: "different origins",
+	}, {
+		desc:             "mismatch targets",
+		prefix:           &gpb.Path{Target: "t1"},
+		suffix:           &gpb.Path{Target: "t2"},
+		wantErrSubstring: "different targets",
+	}}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			got, err := JoinPaths(tt.prefix, tt.suffix)
+			if diff := errdiff.Substring(err, tt.wantErrSubstring); diff != "" {
+				t.Errorf("JoinPaths(%v, %v) got unexpected error diff: %s", tt.prefix, tt.suffix, diff)
+			}
+			if err != nil {
+				return
+			}
+			if diff := cmp.Diff(tt.want, got, cmp.Comparer(proto.Equal)); diff != "" {
```

Comment:
optional: Can use `protocmp.Transform()` instead of proto.Equal. It provides a better diff message since it can traverse the proto message.

Reply:
Done. I had copied the style of the other unit tests. Probably worth doing a larger cleanup at some point to move everything to protocmp.Transform().